### PR TITLE
Fix warning on non-finite nadir point

### LINKED ISSUE #60: KirlikSayin with unbounded objectives
@xgandibleux says:

> I have prepared an exercise for my students which is a project planning 
> problem with 3 objectives, all the variables are discrete (ps: setting 
> the variables t_i >=0 is enough), see the attached document.
> 
> When I solve it with MOA.Lexicographic(all_permutations = true) I get a 
> result (ps: here I have to check the implementation because line 2 and 3 
> in the solution returned should be exchanged because we expect to get 
> the minimum on fct2 when fct2 is minimized, which is not the case - but 
> it is late now, maybe I don't see the obvious-).
> 
> When MOA.KirlikSayin() is called for solving this problem, a warning 
> appears and the resolution is stopped: "objective 1 does not have a 
> finite domain". If I am not wrong the initial conditions for applying KS 
> on this problem are fulfilled, right (objectives have bounded values). 
> Do you have any idea why this warning is raised and then the resolution 
> stopped?

Reproducible example
```Julia
julia> using JuMP

julia> import MultiObjectiveAlgorithms as MOA

julia> import HiGHS

julia> model = Model(() -> MOA.Optimizer(HiGHS.Optimizer))
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: EMPTY_OPTIMIZER
Solver name: MOA[algorithm=MultiObjectiveAlgorithms.Lexicographic, optimizer=HiGHS]

julia> set_silent(model)

julia> set_attribute(model, MOA.Algorithm(), MOA.KirlikSayin())

julia> @variable(model, x >= 0, Int)
x

julia> @variable(model, y >= 0, Int)
y

julia> @constraint(model, x + y >= 1)
x + y ≥ 1.0

julia> @objective(model, Min, 1.0 * [x, y])
2-element Vector{AffExpr}:
 x
 y

julia> optimize!(model)
┌ Warning: Unable to solve problem using `KirlikSayin()` because objective 1 does not have a finite domain.
└ @ MultiObjectiveAlgorithms ~/.julia/packages/MultiObjectiveAlgorithms/RdHL3/src/algorithms/KirlikSayin.jl:126
```

### PR DESCRIPTION
**Pull request by odow**
Closes #60 

From a quick survey of some papers, it seems that computing the nadir point is non-trivial in the general case of N>2 objectives, so let's just give the user a more detailed warning of what went wrong and what they should do about it.

### CODE REVIEW

File: src/MultiObjectiveAlgorithms.jl [hunk 1]
```diff
@@ -597,6 +597,21 @@ function _is_scalar_status_optimal(model::Optimizer)
     return _is_scalar_status_optimal(status)
 end
 
+function _warn_on_nonfinite_nadir(algorithm, sense, index)
```
Comment (remi-garcia) [inline src/MultiObjectiveAlgorithms.jl:600]:
```suggestion
function _warn_on_nonfinite_anti_ideal(algorithm, sense, index)
```

File: src/MultiObjectiveAlgorithms.jl [hunk 2]
```diff
@@ -597,6 +597,21 @@ function _is_scalar_status_optimal(model::Optimizer)
...
+    alg = string(typeof(algorithm))
+    direction = sense == MOI.MIN_SENSE ? "above" : "below"
+    bound = sense == MOI.MIN_SENSE ? "upper" : "lower"
+    @warn(
+        "Unable to solve the model using the `$alg` algorithm because the " *
+        "nadir point of objective $index is not bounded $direction, and the " *
```
Comment (remi-garcia) [inline src/MultiObjectiveAlgorithms.jl:606]:
```suggestion
        "anti-ideal point of objective $index is not bounded $direction, and the " *
```

File: src/algorithms/DominguezRios.jl [hunk 3]
```diff
@@ -178,10 +178,7 @@ function optimize_multiobjective!(algorithm::DominguezRios, model::Optimizer)
         MOI.optimize!(model.inner)
         status = MOI.get(model.inner, MOI.TerminationStatus())
         if !_is_scalar_status_optimal(status)
-            @warn(
-                "Unable to solve problem using `DominguezRios()` because " *
-                "objective $i does not have a finite domain.",
-            )
+            _warn_on_nonfinite_nadir(algorithm, sense, i)
```
Comment (remi-garcia) [inline src/algorithms/DominguezRios.jl:181]:
```suggestion
            _warn_on_nonfinite_anti_ideal(algorithm, sense, i)
```

File: src/algorithms/KirlikSayin.jl [hunk 4]
```diff
@@ -123,10 +123,7 @@ function optimize_multiobjective!(algorithm::KirlikSayin, model::Optimizer)
         MOI.optimize!(model.inner)
         status = MOI.get(model.inner, MOI.TerminationStatus())
         if !_is_scalar_status_optimal(status)
-            @warn(
-                "Unable to solve problem using `KirlikSayin()` because " *
-                "objective $i does not have a finite domain.",
-            )
+            _warn_on_nonfinite_nadir(algorithm, sense, i)
```
Comment (remi-garcia) [inline src/algorithms/KirlikSayin.jl:126]:
```suggestion
            _warn_on_nonfinite_anti_ideal(algorithm, sense, i)
```

**Review by remi-garcia — COMMENTED** [review]
KirlikSayin and DominguezRios do not use the nadir point but the anti-ideal point and an "estimation of the nadir point", respectively.

If the Pareto front is bounded, the nadir point (or an estimation of the nadir point) is also bounded. However the anti-ideal might not be.

In both implementations here, the anti-ideal point is computed hence the warning is necessary.

File: src/MultiObjectiveAlgorithms.jl [hunk 5]
```diff
@@ -597,6 +597,21 @@ function _is_scalar_status_optimal(model::Optimizer)
...
+    alg = string(typeof(algorithm))
+    direction = sense == MOI.MIN_SENSE ? "above" : "below"
+    bound = sense == MOI.MIN_SENSE ? "upper" : "lower"
+    @warn(
+        "Unable to solve the model using the `$alg` algorithm because the " *
+        "anti-ideal point of objective $index is not bounded $direction, and the " *
+        "algorithm requires a finitely bounded objective domain. The easiest " *
+        "way to fix this is to add objective $index as a constraint with a " *
+        "finite $bound. Alteratively, ensure that all of your decision " *
+        "have finite lower and upper bounds."
```
Comment (odow) [inline src/MultiObjectiveAlgorithms.jl:610]:
```suggestion
        "variables have finite lower and upper bounds."
```

**Pull request merged**